conjur/credentials_from_file.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+
+"""
+CredentialsFromFile module
+
+This module holds the logic for writing user-specific credentials
+to a netrc file on the user's machine when they login
+"""
+
+# Builtins
+import logging
+import netrc
+import os
+import stat
+
+# Internals
+from conjur.constants import DEFAULT_NETRC_FILE
+
+class CredentialsFromFile:
+    """
+    CredentialsFromFile
+
+    This class holds logic when credentials are kept in the netrc
+    """
+    @classmethod
+    def save(cls, credential_data):
+        """
+        Method that writes user data to a netrc file
+        and updates permissions on the file
+        """
+        with open(DEFAULT_NETRC_FILE, 'w') as netrc_file:
+            # pylint: disable=line-too-long
+            netrc_file.write(f"machine {credential_data.machine}\nlogin {credential_data.login}\npassword {credential_data.password}\n")
+            netrc_file.close()
+
+        # Ensures that the netrc file is only available to owner
+        os.chmod(DEFAULT_NETRC_FILE, stat.S_IRWXU)
+
+    @classmethod
+    def remove(cls):
+        """
+        Method that removes netrc data from the file.
+        Triggered during a logout
+        """
+        if os.path.exists(DEFAULT_NETRC_FILE) and os.path.getsize(DEFAULT_NETRC_FILE) != 0:
+            open(DEFAULT_NETRC_FILE, "w").close()
+        else:
+            raise Exception("Please log in")
+
+    @classmethod
+    def load(cls):
+        """
+        Method that loads the netrc data.
+        Triggered before each CLI action
+        """
+        # pylint: disable=logging-fstring-interpolation
+        logging.info(f"Using netrc as credential store {DEFAULT_NETRC_FILE}")
+        loaded_netrc = {}
+
+        netrc_obj = netrc.netrc(DEFAULT_NETRC_FILE)
+        # Get the host url from the netrc object
+        netrc_host_url = list(netrc_obj.hosts)[0]
+        netrc_auth = netrc_obj.authenticators(netrc_host_url)
+        login_id, _, api_key = netrc_auth
+
+        loaded_netrc['api_key'] = api_key
+        loaded_netrc['login_id'] = login_id
+
+        return loaded_netrc
```

Comment:
Why returning hash and not `CredentialsData`?
It doesn't make sense.
If you need more data object types like `CredentialsData`, create them.

Reply:
@mbenita-Cyberark also had this suggestion which is 💯 . This was purposeful because I don't have to change Client because users can still supply this info via client and not directly go through the CLI. In short, this is a bigger change then it seems

Reply:
your call.